### PR TITLE
Center/right alignment

### PR DESCRIPTION
**Pull request by sevenshamrocks**
I believe I've isolated the center/right alignment handling. I'm not sure why we need the isRightAligned check. Aren't we closing the paragraph regardless?

### CODE REVIEW

File: addon/gdc.gs [hunk 1]
```diff
@@ -231,8 +228,8 @@ gdc.mdMarkup = {
   // Font changes
   codeOpen:    '`',
   codeClose:   '`',
-  italicOpen:  '*',
```
Comment (evbacher) [inline addon/gdc.gs]: keep from main.

File: addon/gdc.gs [hunk 2]
```diff
@@ -231,8 +228,8 @@ gdc.mdMarkup = {
...
-  italicClose: '*',
+  italicOpen:  '_',
```
Comment (evbacher) [inline addon/gdc.gs]: this is old.

File: addon/gdc.gs [hunk 3]
```diff
@@ -827,7 +819,6 @@ gdc.handleText = function(textElement) {
     // Open attributes (in alphabetical order).
     // Open bold.
     if (!gdc.isBold && bold) {
-      // Check for leading or trailing space.
```
Comment (evbacher) [inline addon/gdc.gs]: comment for a fix in progress

File: addon/gdc.gs [hunk 4]
```diff
@@ -850,26 +841,39 @@ gdc.handleText = function(textElement) {
     }
     // Open italic.
     if (!gdc.isItalic && italic) {
-      // Check for leading or trailing space.
```
Comment (evbacher) [inline addon/gdc.gs]: comment for a fix in progress

File: addon/gdc.gs [hunk 5]
```diff
@@ -850,26 +841,39 @@ gdc.handleText = function(textElement) {
...
       gdc.isItalic = true;
       gdc.openAttrs.push(gdc.italic);
       gdc.writeStringToBuffer(gdc.markup.italicOpen);
     }
     // Open strikethrough.    
     if (!gdc.isStrikethrough && strikethrough) {
-      // Check for leading or trailing space.
```
Comment (evbacher) [inline addon/gdc.gs]: comment for a fix in progress

File: addon/gdc.gs [hunk 6]
```diff
@@ -1003,17 +1012,9 @@ gdc.isBullet = function(glyphType) {
 };
 
 // Sets appropriate markup object.
-gdc.useMarkdown = function() {
+gdc.useMarkdown = function(){
   gdc.isHTML = false;
   gdc.markup = gdc.mdMarkup;
-
```
Comment (evbacher) [inline addon/gdc.gs]: keep this block for italic/bold markup (from latest main).

File: addon/gdc.gs [hunk 7]
```diff
@@ -1641,10 +1644,7 @@ gdc.topComment = '<!-----\n\n'
 ;
   
 md.doMarkdown = function(config) {
-  // Call config first!
   gdc.config(config);
-  gdc.useMarkdown();
-
```
Comment (evbacher) [inline addon/gdc.gs]: keep this block (with gdc.config(config) first.

File: addon/gdc.gs [hunk 8]
```diff
@@ -1666,10 +1666,10 @@ md.doMarkdown = function(config) {
   gdc.info = '\n\nConversion time: ' + eTime + ' seconds.\n' + gdc.info;
   
   // Note ERRORs or WARNINGs or ALERTs at the top if there are any.
-  gdc.errorSummary = '';
```
Comment (evbacher) [inline addon/gdc.gs]: keep this version of blank errorSummary.

File: addon/gdc.gs [hunk 9]
```diff
@@ -1666,10 +1666,10 @@ md.doMarkdown = function(config) {
...
+  gdc.errorSummary = 'Yay, no errors, warnings, or alerts!'
   if ( gdc.errorCount || gdc.warningCount || gdc.alertCount ) {
     gdc.errorSummary = 'You have some errors, warnings, or alerts. '
-      + 'If you are using reckless mode, turn it off to see useful information and inline alerts.'
```
Comment (evbacher) [inline addon/gdc.gs]: keep main version.

File: addon/gdc.gs [hunk 10]
```diff
@@ -1690,7 +1690,7 @@ md.doMarkdown = function(config) {
   // Add info comment if desired.
   if (!gdc.suppressInfo) {
     gdc.out = gdc.info + '\n----->\n\n' + gdc.out;
-  } else if (gdc.suppressInfo && gdc.errorSummary !== '') {
```
Comment (evbacher) [inline addon/gdc.gs]: keep main version

File: addon/gdc.gs [hunk 11]
```diff
@@ -1802,14 +1802,17 @@ md.handleParagraph = function(para) {
   gdc.inHeading = false;
   gdc.state.isMixedCode = false;
   gdc.numChildren = para.getNumChildren();
-  // Do not bother with empty paragraphs (blank lines). (Except we preserve them for code blocks.)
+  // Preserve blank lines in body text as well as code blocks
```
Comment (evbacher) [inline addon/gdc.gs]: keep your version

File: addon/gdc.gs [hunk 12]
```diff
@@ -1879,8 +1881,19 @@ md.handleParagraph = function(para) {
       // do nothing: we also want to not add the tablePrefix.
       // But check for table cell or definition list.
     } else if (!gdc.startingTableCell && !gdc.inDlist) {
-      gdc.writeStringToBuffer(gdc.markup.pOpen);
-    }
+      // This is where we want to check for right/center alignment so that the proper paragraph style can be applied.
```
Comment (evbacher) [inline addon/gdc.gs:1883]: keep this version

File: addon/gdc.gs [hunk 13]
```diff
@@ -1901,13 +1914,6 @@ md.handleParagraph = function(para) {
     }
   }
 
-  // Check horizontal alignment. We can style right alignment using an HTML paragraph.
```
Comment (evbacher) [inline addon/gdc.gs:1904]: keep right alignment for Markdown (and HTML). Your version doesn't handle right-aligned text for Markdown.

**Review by evbacher — CHANGES_REQUESTED** [review]
It looks like your branch doesn't have a lot of changes from the latest main (or the latest dev-main).

I'd like to merge your changes to latest dev-main and test. Can you check my review comments and make changes?

**Comment by sevenshamrocks** [comment]
I think this is an issue with main vs. main- 1.038 base. I have the removed lines in my branch from what I can see. When compared to main, many of these issues are resolved.  

Part of it may be that I don't fully understand the comparisons and pull request direction.

File: addon/gdc.gs (hunk 13 above)
Comment (sevenshamrocks) [reply addon/gdc.gs:1904]: So would we want !gdc.isHTML in the if statement?

Comment (sevenshamrocks) [reply addon/gdc.gs:1904]: Or rather, we can modify the if statement on 1884 to remove the gdc.isHTML check?

**Comment by sevenshamrocks** [comment]
Oh. I see what the issue is. The first commit was using the old code. Subsequent commits should be updated with all changes from main. You should only see differences with the text-align codes when comparing with all commits on this pull.

**Review by evbacher — APPROVED** [review]
Looks good. I'm going to approve this and test the dev branch code. Onward!

Will get to the other pull requests eventually -- I have a busy couple of days but will try to fit some. of this in.

**Pull request merged**